internal/kernel_SSE.h

```diff
@@ -0,0 +1,332 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// kernel_SSE.h: a collection of Intel SSE optimized kernels.
+// Check in kernel_default.h which one(s) are actually used by default.
+// Others are mere experiments; they are still covered by tests
+// in case they might be useful some day.
+// Writen by: Sagi Marovich, mail: sagi.marcovich@intel.com
+
+#ifndef GEMMLOWP_INTERNAL_KERNEL_SSE_H_
+#define GEMMLOWP_INTERNAL_KERNEL_SSE_H_
+
+
+#include "kernel.h"
+#include <cassert>
+#include <string.h>
+#include <iostream>
+
+
+
+
+namespace gemmlowp {
+
+#ifdef GEMMLOWP_SSE32
```

Comment:
Since there are multiple versions of SSE, please explictly specify which one is being targeted. Here I see pmovzxbw which is a SSE 4 instruction, so maybe GEMMLOWP_SSE4_32BIT   (I know I used GEMMLOWP_NEON32 but evidently this naming scheme doesn't scale well here so we need to change it).

Also, such GEMMLOWP_SSE* macros need to be defined automatically in common.h based on compiler predefined symbols, so that SSE paths are automatically picked up. See e.g. what Eigen does here:
https://bitbucket.org/eigen/eigen/src/a2727d3da0c40c0781e515154cc0828060d13f3a/Eigen/Core?at=default#Core-115